log4j-api/src/main/java/org/apache/logging/log4j/status/StatusLogger.java

```diff
@@ -251,7 +276,13 @@ private static Level readFallbackListenerLevel(final Properties fileProvidedProp
 
         private static DateTimeFormatter readInstantFormatter(final Properties fileProvidedProperties) {
             final String format = readProperty(fileProvidedProperties, STATUS_DATE_FORMAT);
-            return format != null ? DateTimeFormatter.ofPattern(format) : null;
+            if (format == null) {
+                return null;
+            }
+            final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
+            final String zoneId = readProperty(fileProvidedProperties, STATUS_DATE_FORMAT_ZONE);
+            final ZoneId effectiveZoneId = zoneId != null ? ZoneId.of(zoneId) : ZoneId.systemDefault();
```

Comment:
What happens if `ZoneId.of` throws an exception?

Reply:
I've extensively improved such checks along with tests. Could you check it again, please?